## Importing libraries

In [1]:
import pandas as pd
import numpy as np

## Importing data file

In [2]:
data = pd.read_csv(r"C:\Users\ank\Desktop\Python\Multiple-Linear-Regression-master\Multiple-Linear-Regression-master\50_Startups.csv")

In [3]:
data.head()

R&D Spend  Administration  Marketing Spend       State     Profit
0  165349.20       136897.80        471784.10    New York  192261.83
1  162597.70       151377.59        443898.53  California  191792.06
2  153441.51       101145.55        407934.54     Florida  191050.39
3  144372.41       118671.85        383199.62    New York  182901.99
4  142107.34        91391.77        366168.42     Florida  166187.94

## Creating two variables

In [4]:
X = data.iloc[:,:-1]
y = data.iloc[:,-1]

## Converting categorical column to different columns

In [5]:
state = pd.get_dummies(data['State'],drop_first = True)

In [6]:
X = pd.concat([X,state], axis= 1)

In [7]:
X = X.drop('State', axis = 1)

## Splitting dataset into train set and test set

In [10]:
from sklearn.model_selection import train_test_split
X_train, X_test, y_train, y_test = train_test_split(X,y, test_size = 0.2, random_state = 2)

## Fitting multiple linear regression to training set

In [12]:
from sklearn.linear_model import LinearRegression
regression = LinearRegression()
regression.fit(X_train,y_train)

LinearRegression()

## Predicting test values

In [13]:
y_pred = regression.predict(X_test)

In [16]:
from sklearn.metrics import r2_score
score = r2_score(y_test,y_pred)

In [17]:
score

0.9783180289265575

### So our models accuracy for prediction of price is 97.83%